### ISSUE TITLE
When missing EPAR table, just emit INFO alert

### ISSUE THREAD

**Issue by khaledhosny** [googlefonts/fontbakery#818]
No idea what is `EPAR` table is (not a standard table AFAICT) and why my fonts should have it.

**Comment by davelab6** [comment]
See https://davelab6.github.io/epar/

Perhaps @dberlow can make the case. 

It is/was a way of expressing common licensing permissions and restrictions in metadata; while almost nothing supported it, I wonder that adding it to everything in GF could help make it more popular.

**Comment by khaledhosny** [comment]
I see, but even fonttools does not support it. I think an ERROR is just too strong.

**Comment by davelab6** [comment]
I agree. I update this issue to be actionable

**Comment by dberlow** [comment]
Dave and Khaled,

I agree it is rather strong of the Fontbakery to require epar.

To correct Dave's memory, Epar is a way of expressing electronic permissions and recommendations. There are no "restrictions" in epar. 

For Google, permissions would be a dull place, compared to commercial, fonts, but

For Google, recommendations would be a hoppin' place, with all kinds of per style data that could be useful in a catalog, or to a composition app, as well as being presented to the user. Much finer values for a style's weight and width, than are in OS/2, ar in epar, so e.g. This has allowed us to define widths and weights relatively with great precision to particular contrasts beyond regular and bold, and between the widths of various sizes. http://store.typenetwork.com/foundry/fontbureau/fonts/agency-fb?layout=overview
Is an example of this in action, where none of the TN "overview" pages have a single manual style of size specification, running instead on an algorithm based on accurate 1/1000ths of an em values for weight and width, along with optical size master data.

 (If you don't own OT, like us, and you need to move along, like us, this stuff is becoming handier in one place.)

Cheers,

db

**Comment by davelab6** [comment]
> There are no "restrictions" in epar.

Thanks for correcting me on this :) Its how I remember hearing about it for the first time, I guess...

> > For instance, if the EEULAA says "no
> > modification", then a message might appear that says "The license
> > for this font does not allow user modification. Please contact the
> > vendor to upgrade your license." and the font would not be opened.

- https://sourceforge.net/p/fontforge/mailman/message/6164525/

But I agree that Recommendations are great; perhaps we can distinguish between EPAR and EEULAA  :)

**Comment by dberlow** [comment]
Sadly, because of the DRM knee jerk society, and since the initial failure to describe epar as Anything else...  It has been better to selfishly benefit from it on our own, then to continually explain its benefits to others, all far more needy than us.

Good luck!

**Comment by felipesanches** [comment]
I did change the test to emit an INFO result for fonts that lack an EPAR table. It was very weird to emit PASS in both cases. An INFO makes much more sense here.

See: https://github.com/googlefonts/fontbakery/pull/1476/commits/9a09133d910d5ef0d28cc92798144fcc81b4af64

**Comment by davelab6** [comment]
That's cool

**Comment by timboudreau** [comment]
I am generating a font using Java code, not using any visual tool - reviving and modernizing a bunch of ancient code extracted from the circa 2004 Doubletype font editor.  I have managed to add writers for newer tables, and the results work nicely, and fontbakery tests largely pass.

If the goal of emitting this INFO alert is to popularize use of the table, it would be truly helpful if, somewhere on the planet, the actual structure of the table were clearly and concisely documented.

I have looked at https://davelab6.github.io/epar/ which seems to contain a bunch of discussion of things that might or might not be implemented for all sorts of media formats - but no _this thing is a `uint16` followed by that thing which is a pascal string_ that would let anyone actually implement this thing.  There are a couple of links which look like they might have once linked to further info, but now are squatted advertising sites.
